### ISSUE TITLE
0.12 Networking

### ISSUE THREAD

**Issue by RandomAltThing** [DragonetMC/Dragonet#61]
We need to implement packets for 0.12.0, as well as enchanting, anvils, and many other things.

I was wondering if @jython234 Would help with the protocal?

**Comment by jython234** [comment]
Yes I can help.

On Wednesday, August 5, 2015, RandomAltThing <notifications@github.com>
wrote:

> We need to implement packets for 0.12.0, as well as enchanting, anvils,
> and many other things.
>
> I was wondering if @jython234 <https://github.com/jython234> Would help
> with the protocal?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61>.
>


--

**Comment by RandomAltThing** [comment]
@jython234 Its funny how no one helps my projects -_-

**Comment by jython234** [comment]
Don't worry not many people help my projects either.

On Wednesday, August 5, 2015, RandomAltThing <notifications@github.com>
wrote:

> @jython234 <https://github.com/jython234> Its funny how no one helps my
> projects -_-
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128052533>.
>


--

**Comment by RandomAltThing** [comment]
@jython234 But you're skilled enough to do it yourself whereas I am not. I had some one help me with raknet protocal (they gave me the code and I modified it to fit my project which can be found [here](http://github.com/MineServerProject/Mineserver)

I can help you on your projects if you like

**Comment by jython234** [comment]
It's just that I can't send chunks :(

On Wednesday, August 5, 2015, RandomAltThing <notifications@github.com>
wrote:

> @jython234 <https://github.com/jython234> But you're skilled enough to do
> it yourself whereas I am not. I had some one help me with raknet protocal
> (they gave me the code and I modified it to fit my project which can be
> found here <https://github.com/MineServerProject/Mineserver>
>
> I can help you on your projects if you like
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128073464>.
>


--

**Comment by RandomAltThing** [comment]
@jython234 Can you help on my project literally just the raknet code you don't have to do the mcpe packets

**Comment by jython234** [comment]
@RandomAltThing This is not a forum
https://github.com/BlockServerProject/JRakLib/wiki/Getting%20Started

**Comment by jython234** [comment]
@DragonetMC/developers I noticed that the networking is actually just a PC-PE packet translator. Is it planed to stay this way?

**Comment by TekkitCommando** [comment]
Umm @jython234 thats really up to @DefinitlyEvil

**Comment by jython234** [comment]
@TekkitCommando That's why i'm asking :)

**Comment by RandomAltThing** [comment]
And up to @TheMCPEGamer

**Comment by TekkitCommando** [comment]
Oh you did at developers so I thought you were asking the contributors xD

**Comment by RandomAltThing** [comment]
You are in the developers team @TekkitCommando

**Comment by TekkitCommando** [comment]
I know and that's why I thought he was asking me (one of the contributers)

**Comment by DefinitlyEvil** [comment]
@jython234 Yes, I think there should be a translator because some code is compatible with Glowstone++'s code so we don't have to copy them over. ;)

**Comment by jython234** [comment]
Ok good idea. I have to admit that there is a problem with joining :/. It
appears it takes too long for the server to handle the login, and then my
client disconnects.

On Wednesday, August 5, 2015, The Evil Man <notifications@github.com> wrote:

> @jython234 <https://github.com/jython234> Yes, I think there should be a
> translator because some code is compatible with Glowstone++'s code so we
> don't have to copy them over. ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128199258>.
>


--

**Comment by DefinitlyEvil** [comment]
@jython234 It's Glowstone++'s problem I think, it generates structures when player joining. And I will do the networking update job.

**Comment by mastercoms** [comment]
The fix was made available in https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/commit/360ad4132d855ce1a12b4d4b74e90447a4e6b136 and was made default in https://github.com/GlowstonePlusPlus/GlowstonePlusPlus/commit/38df536bf2ce50145a9e02a162aa852756297c97

**Comment by DefinitlyEvil** [comment]
@mastercoms Will merge manually soon. ;)

**Comment by RandomAltThing** [comment]
@DragonetMC/developers Please push all 0.12.0 related commits to the **mcpe-0.12** branch where we can merge later.

Thanks.

**Comment by DefinitlyEvil** [comment]
@RandomAltThing Already committed to master branch. xD

**Comment by jython234** [comment]
I will push JRakLib to JRakLib-integration.

On Thursday, August 6, 2015, The Evil Man <notifications@github.com> wrote:

> @RandomAltThing <https://github.com/RandomAltThing> Already committed to
> master branch. xD
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128293982>.
>


--

**Comment by DefinitlyEvil** [comment]
@jython234 okay.

**Comment by DefinitlyEvil** [comment]
@jython234 I renamed it to jraklib.

**Comment by jython234** [comment]
@DefinitlyEvil that's fine :) my IDE is very confused as there are ghost
methods like getSession() which don't exist, yet they compile and run fine
:/

On Friday, August 7, 2015, The Evil Man <notifications@github.com> wrote:

> @jython234 <https://github.com/jython234> I renamed it to jraklib.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128605414>.
>


--

**Comment by jython234** [comment]
@DefinitlyEvil NVM Discovered this: https://github.com/mplushnikov/lombok-intellij-plugin

**Comment by DefinitlyEvil** [comment]
@jython234 You should use NetBeans. ;)

**Comment by TekkitCommando** [comment]
@DefinitlyEvil Intellij and eclipse is the only ide's I trust xD

**Comment by jython234** [comment]
I use IntelliJ, by far best IDE.

On Friday, August 7, 2015, TekkitCommando <notifications@github.com> wrote:

> @DefinitlyEvil <https://github.com/DefinitlyEvil> Intellij and eclipse is
> the only ide's I trust xD
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128877607>.
>


--

**Comment by TekkitCommando** [comment]
Yeah pretty freakin great

**Comment by mastercoms** [comment]
Me too.

**Comment by jython234** [comment]
So who does what? The only perso I know here is @DefinitlyEvil. I know he
does networking, and he started the project right? But what do you guys
do/work on?

On Saturday, August 8, 2015, mastercoms <notifications@github.com> wrote:

> Me too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128984275>.
>


--

**Comment by DefinitlyEvil** [comment]
@mastercoms They are from @GlowstonePlusPlus , they helps to fix MCPC bugs around. ;)

**Comment by TekkitCommando** [comment]
You Mean @jython234 ? XD

**Comment by RandomAltThing** [comment]
@Jython234 I was originally here to work on he android app with is now discontinued. However, I may add some stuff its just hard to work with because most API stuff is already done from bukkit and GlowstonePlusPlus. I like making my own API -_-

**Comment by TekkitCommando** [comment]
Spigot is widely used it would be a terrible idea to create a new api cuz less people would want to use it

**Comment by jython234** [comment]
Oh ok. But there should be a better system to sync with the glow stone
repository.

On Saturday, August 8, 2015, TekkitCommando <notifications@github.com>
wrote:

> Spigot is widely used it would be a terrible idea to create a new api cuz
> less people would want to use it
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-128992715>.
>


--

**Comment by TekkitCommando** [comment]
The reason why it's so hard to merge Glowstone changes is because how the dragonet repo is setup

**Comment by mastercoms** [comment]
Yes, but I guess out of preference/habit, @DefinitlyEvil doesn't want to change the repo structure.

**Comment by jython234** [comment]
It was a pain for me to set it up in the IDE. The structure needs to be
changed.

On Saturday, August 8, 2015, mastercoms <notifications@github.com> wrote:

> Yes, but I guess out of preference/habit, @DefinitlyEvil
> <https://github.com/DefinitlyEvil> doesn't want to change the repo
> structure.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-129012624>.
>


--

**Comment by mastercoms** [comment]
We've suggested it before, but it has been ignored or rejected.

**Comment by jython234** [comment]
@mastercoms Heh mysteriously.

**Comment by TekkitCommando** [comment]
lol

**Comment by DefinitlyEvil** [comment]
Well, I don't want to change the structure because I am bad at using Git tools, I am afraid that I might break the repo data.

**Comment by RandomAltThing** [comment]
@DefinitlyEvil Well why do you think we hired @jython234 for?

**Comment by jython234** [comment]
Eh?

On Sunday, August 9, 2015, RandomAltThing <notifications@github.com> wrote:

> @DefinitlyEvil <https://github.com/DefinitlyEvil> Well why do you think
> we hired @jython234 <https://github.com/jython234> for?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-129143363>.
>


--

**Comment by DefinitlyEvil** [comment]
@jython234 He is coding BlockServer and he has expirence with MCPE so I think he can help us. In a word, the more the better.

**Comment by jython234** [comment]
@DefinitlyEvil You forget the fact I never got the client to spawn with
chunks. But I have been working with MCPE since late 2013, so that's good.

On Sunday, August 9, 2015, The Evil Man <notifications@github.com> wrote:

> @jython234 <https://github.com/jython234> He is coding BlockServer and he
> has expirence with MCPE so I think he can help us. In a word, the more the
> better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-129197261>.
>


--

**Comment by DefinitlyEvil** [comment]
@jython234 xD I wasn't wrong, you have the expirence although it's bad expirence. lol...

**Comment by jython234** [comment]
It's not bad, I've spawned with chunks one before, that was my MCBridge
plugin that is similar to dragonet. I discontinued it because the structure
was horrible.

On Sunday, August 9, 2015, The Evil Man <notifications@github.com> wrote:

> @jython234 <https://github.com/jython234> xD I wasn't wrong, you have the
> expirence although it's bad expirence. lol...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-129197652>.
>


--

**Comment by DefinitlyEvil** [comment]
@jython234 okay. xD

**Comment by jython234** [comment]
But yeah if I'm going to help, I need to know the structure and what
glowkit is.

On Sunday, August 9, 2015, The Evil Man <notifications@github.com> wrote:

> @jython234 <https://github.com/jython234> okay. xD
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-129197745>.
>


--

**Comment by TekkitCommando** [comment]
glowkit is Glowstone++'s Continued Version of Bukkit

**Comment by mastercoms** [comment]
Basically, Glowstone used Bukkit, but when the Bukkit developers quit, we had to make our own continuation of the API. I came up with the name Glowkit, and it stuck. Of course, Spigot has continued the API, so we've switched to using mostly Spigot's API, with a few of our own enhancements and PaperSpigot-API.

**Comment by jython234** [comment]
Oh ok. But why do we have it as source here? Can't we add it as maven
dependency?

On Sunday, August 9, 2015, mastercoms <notifications@github.com> wrote:

> Basically, Glowstone used Bukkit, but when the Bukkit developers quit, we
> had to make our own continuation of the API. I came up with the name
> Glowkit, and it stuck. Of course, Spigot has continued the API, so we've
> switched to using mostly Spigot's API, with a few of our own enhancements
> and PaperSpigot-API.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-129199275>.
>


--

**Comment by mastercoms** [comment]
https://www.youtube.com/watch?v=t348e24vDyA

**Comment by jython234** [comment]
Great.

On Sunday, August 9, 2015, mastercoms <notifications@github.com> wrote:

> Medic's answer to every question in the universe.
> <https://www.youtube.com/watch?v=t348e24vDyA>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DragonetMC/Dragonet/issues/61#issuecomment-129260536>.
>


--

**Comment by DefinitlyEvil** [comment]
That folder is a git subtree(I don't know how to use submodule).
